syntax = "proto3";

package ctp;

//数据类型MSG_DATA_FUTURE
message TDF_FUTURE_DATA
{
	string        szWindCode = 1;         //600001.SH
	string        szCode = 2;             //原始Code
	int32         nActionDay = 3;             //业务发生日(自然日)
	int32         nTradingDay = 4;            //交易日
	int32			 nTime = 5;					//时间(HHMMSSmmm)
	int32			 nStatus = 6;				//状态
	int64		 iPreOpenInterest = 7;		//昨持仓
	fixed32 nPreClose = 8;				//昨收盘价
	fixed32 nPreSettlePrice = 9;		//昨结算
	fixed32 nOpen = 10;					//开盘价
	fixed32 nHigh = 11;					//最高价
	fixed32 nLow = 12;					//最低价
	fixed32 nMatch = 13;				//最新价
	int64		 iVolume = 14;				//成交总量
	int64		 iTurnover = 15;				//成交总金额
	int64		 iOpenInterest = 16;			//持仓总量
	fixed32 nClose = 17;				//今收盘
	fixed32 nSettlePrice = 18;			//今结算
	fixed32 nHighLimited = 19;			//涨停价
	fixed32 nLowLimited = 20;			//跌停价
	int32			 nPreDelta = 21;			    //昨虚实度
	int32			 nCurrDelta = 22;            //今虚实度
	repeated fixed32 nAskPrice = 23;			//申卖价
	repeated fixed32 nAskVol = 24;			//申卖量
	repeated fixed32 nBidPrice = 25;			//申买价
	repeated fixed32 nBidVol = 26;			//申买量
	//Add 20140606
	int32	nAuctionPrice = 27;			    //波动性中断参考价
	int32	nAuctionQty = 28;			    //波动性中断集合竞价虚拟匹配量
	int32 nAvgPrice = 29;                      //郑商所期货均价
};
